src/Native/include/nncase/ntt/kernels/gather.h

```diff
@@ -19,110 +19,146 @@
+        ranked_shape<rank> src_index;
+        for (size_t i = 0; i < rank; i++) {
+            src_index[i] = 0;
+        }
 
-    for (size_t i = 1; i < arrSize; ++i) {
-        if (arr[i] == arr[i - 1] + 1) {
-            ++length;
+        if (input_conti_dims == rank && count != indices_len) {
+            apply(domain_before_axis, [&](auto index) {
+                for (size_t i = 0; i < count; i++) {
+                    auto seq = segments[i];
+                    for (size_t j = 0; j < Axis; j++) {
+                        src_index[j] = index[j];
+                    }
+                    src_index[Axis] = indices.elements()[seq.start];
+                    auto len =
+                        seq.length * domain_after_axis.length() * element_size;
+                    std::memcpy(addr_output_byte, &(input(src_index)), len);
+                    addr_output_byte += len;
+                }
+            });
+        } else if (input_conti_dims == rank) {
+            apply(domain_before_axis, [&](auto index) {
+                for (size_t i = 0; i < TB::size(); i++) {
+
+                    for (size_t j = 0; j < Axis; j++) {
+                        src_index[j] = index[j];
+                    }
+                    src_index[Axis] = indices.elements()[i];
+                    auto addr_input = reinterpret_cast<const element_type *>(
+                        &(input(src_index)));
+                    constexpr auto len = domain_after_axis.length();
+
+                    ntt::u_memcpy<element_type>(addr_input, 1,
+                                                addr_output_element, 1, len);
+                    addr_output_element += len;
+                }
+            });
         } else {
-            segments[segmentCount].start = start;
-            segments[segmentCount].length = length;
-            ++segmentCount;
-            start = i;
-            length = 1;
+            apply(out_shape, [&](auto out_index) {
+                // in_index[:axis] = out_index[:axis]
+                loop<Axis>([&](auto i) { in_index[i] = out_index[i]; });
+
+                // in_index[axis] = indices(indices_index)
+                loop<indices_rank>(
+                    [&](auto i) { indices_index[i] = out_index[i + Axis]; });
+                in_index[Axis] = indices(indices_index);
+
+                // in_index[axis:] = out_index[axis:]
+                loop<rank - (Axis + 1)>([&](auto i) {
+                    in_index[Axis + 1 + i] = out_index[Axis + indices_rank + i];
+                });
+                output(out_index) = input(in_index);
+            });
         }
     }
 
-    segments[segmentCount].start = start;
-    segments[segmentCount].length = length;
-    ++segmentCount;
-
-    return segmentCount;
-}
+  private:
+    struct Segment {
+        size_t start;
+        size_t length;
+    };
+
+    template <typename T>
+    size_t findContinuousSegments(const T *arr, size_t arrSize,
```

Comment:
It should be snake naming.

Reply:
Thanks，I have already changed to snake naming.